torch/csrc/jit/codegen/cuda/kernel_cache.h

```diff
@@ -77,74 +77,75 @@ class TORCH_CUDA_API InputsIdLookup {
+//        - GraphCache translates PyTorch IR into Fusion IR and pass it to a
+//          `FusionExecutorCache`;
+//        - GraphCache assumes all inputs to comply with profiling information,
+//          mostly tensor size & contiguity (see note on unique computational
+//          graph). The assumption is assured at runtime by
+//          `prim::CudaFusionGuard`;
+//        - GraphCache handles permutation for I/O tensors, when they share
+//          global stride order. This permutation facilitates dimension
+//          collapsing, which gives simpler indexing.
 //     b. FusionExecutorCache
-//        - holds a group of `FusionExecutor` to handle dynamic shape (varying
-//          tensor sizes)
-//        - currently this is a dummy implementation and has branching to handle
-//          different scheduler for point-wise fusion and reduction fusion;
+//        - has a single `Fusion`, FusionExecutorCache handles kernel schedule
+//          and passed scheduled tensor to `FusionExecutor` to generate code;
+//        - create `FusionExecutor` instances to handle heuristics from dynamic
+//          shape (varying tensor sizes);
+//        - create `FusionExecutor` instances to handle different devices;
+//        - holds input cache `InputsIdLookup`, which allow cache on heuristics
+//          and launch parameters to reduce latency.
 //
-// * note computational graph
+// * note on unique computational graph
 // In theory, computational graph should refer to only the computational nodes
 // in a subgraph and should remain agnostic to input meta info, like
 // shape, strides, type e.t.c.. However, the contract right here is fuzzy.
 // Different executor applies their own protocol of what is a unique
 // computational graph. e.g. Legacy Executor embeds tensor type & dimensionality
 // in the graph, while Profiling Executor keeps symbolic shape as well as stride
 // order in the graph as well.
-// Our definition of computational graph is relaxed to support Legacy Executor,
-// so the `GraphCache` could handle varying memory layout of strided tensor
-// (different stride order & contiguity information). We utilize the profiling
-// information now by generating an entry in GraphCache with the given profiling
-// record.
+//
+// Our definition of a "unique" computational graph is aligned with `Fusion` IR,
+// hence the requirement extends to meta information on input tensors.
+// Which means, for each input tensor, following properties are fixed:
+//     a) stride order;
+//     b) contiguity information;
+//     c) broadcasting semantics (size-1 or not);
+//     d) rank;
+//     e) scalar type;
 
 class FusionExecutorCache {
  public:
   // create new fusion executor cache at a given device to handle kernel
   // generation of dynamic sizes;
   // fusion executor is taking the ownership of `fusion`;
-  FusionExecutorCache(std::unique_ptr<Fusion>&& fusion, at::Device device);
+  FusionExecutorCache(std::unique_ptr<Fusion>&& fusion);
```

Comment:
not critical, but the comments for the public methods can be easily converted to Doxygen by replacing // with //!